packages/js/components/src/experimental-select-control/test/use-async-filter.spec.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import { act, renderHook } from '@testing-library/react-hooks';
+import { useDebounce } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { useAsyncFilter } from '../';
+
+jest.mock( '@wordpress/compose', () => ( {
+	...jest.requireActual( '@wordpress/compose' ),
+	useDebounce: jest.fn( ( cb: CallableFunction ) => cb ),
+} ) );
+
+describe( 'useAsyncFilter', () => {
+	const filter = jest.fn();
+	const onFilterStart = jest.fn();
+	const onFilterEnd = jest.fn();
+	const onFilterError = jest.fn();
+
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'should filter the items successfully', async () => {
+		const filteredItems: string[] = [];
+		filter.mockResolvedValue( filteredItems );
+
+		const { result } = renderHook( () =>
+			useAsyncFilter( {
+				filter,
+			} )
+		);
+
+		const inputValue = '';
```

Comment:
I wonder if it would be better in these tests to set `inputValue` to a non-zero-length string. Because, I could see any implementation bug returning `''`, but it would be less likely for it to return something more unique, like `'my filter input'`.